torch/csrc/jit/python/init.cpp

```diff
@@ -1482,52 +1820,79 @@ void initJITBindings(PyObject* module) {
               "__gt__",
               [](c10::SymIntNode a, py::object b) {
-                auto snb = toSymIntNode(a, b);
-                return a->gt(snb);
+                if (is_int_or_symint(b)) {
+                  auto snb = toSymIntNode(a, b);
+                  return py::cast(a->gt(snb));
+                } else if (is_float_or_symfloat(b)) {
+                  auto snb = toSymFloatNode(a, b);
+                  return py::cast(a->gt(snb));
+                } else {
+                  throw py::reference_cast_error();
+                }
               })
           .def(
               "__lt__",
-              [](c10::SymIntNode a, py::object b) -> c10::SymIntNode {
-                auto snb = toSymIntNode(a, b);
-                return a->lt(snb);
+              [](c10::SymIntNode a, py::object b) -> py::object {
+                if (is_int_or_symint(b)) {
+                  auto snb = toSymIntNode(a, b);
+                  return py::cast(a->lt(snb));
+                } else if (is_float_or_symfloat(b)) {
+                  auto snb = toSymFloatNode(a, b);
+                  return py::cast(a->lt(snb));
+                } else {
+                  throw py::reference_cast_error();
+                }
               })
           .def(
               "__le__",
-              [](c10::SymIntNode a, py::object b) -> c10::SymIntNode {
-                auto snb = toSymIntNode(a, b);
-                return a->le(snb);
+              [](c10::SymIntNode a, py::object b) -> py::object {
+                if (is_int_or_symint(b)) {
+                  auto snb = toSymIntNode(a, b);
+                  return py::cast(a->le(snb));
+                } else if (is_float_or_symfloat(b)) {
+                  auto snb = toSymFloatNode(a, b);
+                  return py::cast(a->le(snb));
+                } else {
+                  throw py::reference_cast_error();
+                }
               })
           .def(
               "__ge__",
-              [](c10::SymIntNode a, py::object b) -> c10::SymIntNode {
-                auto snb = toSymIntNode(a, b);
-                return a->ge(snb);
+              [](c10::SymIntNode a, py::object b) -> py::object {
+                if (is_int_or_symint(b)) {
+                  auto snb = toSymIntNode(a, b);
+                  return py::cast(a->ge(snb));
+                } else if (is_float_or_symfloat(b)) {
+                  auto snb = toSymFloatNode(a, b);
+                  return py::cast(a->ge(snb));
+                } else {
+                  throw py::reference_cast_error();
+                }
               })
           .def(
               "__ceil__",
               [](c10::SymIntNode a) -> c10::SymIntNode { return a->ceil(); })
+          .def(
+              "__floor__",
+              [](c10::SymIntNode a) -> c10::SymIntNode { return a->floor(); })
           .def(
               "__neg__",
               [](c10::SymIntNode a) -> c10::SymIntNode { return a->neg(); })
-          .def(
```

Comment:
The `__min__` and `__max__` are not actually a thing in python so just removing these. Happy to add them back if anyone has some custom logic for using these (but I couldn't find any).